src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1238,6 +1239,44 @@ void CompilationMemoryStatisticDCmd::execute(DCmdSource source, TRAPS) {
   CompilationMemoryStatistic::print_all_by_size(output(), human_readable, minsize);
 }
 
+VMInspectDCmd::VMInspectDCmd(outputStream* output, bool heap) :
+                                     DCmdWithParser(output, heap),
+  _address("address", "", "STRING", true, nullptr),
+  _verbose("-verbose", "", "BOOLEAN", false, "false") {
+
+  _dcmdparser.add_dcmd_argument(&_address);
+  _dcmdparser.add_dcmd_option(&_verbose);
+}
+
+void VMInspectDCmd::execute(DCmdSource source, TRAPS) {
+  DebuggingContext dc{}; // avoid asserts
+
+  if (!UnlockDiagnosticVMOptions) {
```

Comment:
Drive-by comment: As per the comment from @plummercj  this flag should not be hijacked for this purpose. If you need a VM flag to control this feature then please add one. Thanks.